src/core/actions/unitactions/Capture.java

```diff
@@ -23,20 +24,38 @@ public City getTargetCity() {
 
     @Override
     public LinkedList<Action> computeActionVariants(final GameState gs) {
-        //TODO: compute Capture city actions
-        return null;
+
+        LinkedList<Action> captures = new LinkedList<>();
+        if(isFeasible(gs)){
+            Capture c = new Capture(this.unit);
+            c.setTargetCity(this.targetCity);
+            captures.add(c);
+        }
+
+        return captures;
     }
 
     @Override
     public boolean isFeasible(final GameState gs)
     {
         //todo: check if capturing this city is feasible.
-        return false;
+        Board b = gs.getBoard();
+        if(b.getUnitAt(targetCity.getX(),targetCity.getY()) != null)
+            return false;
+        else if(targetCity.getX() > unit.getCurrentPosition().x  + this.unit.RANGE ||targetCity.getY() > unit.getCurrentPosition().y  + this.unit.RANGE)
+            return false;
+
+        return true;
     }
 
     @Override
     public boolean execute(GameState gs) {
         //todo: execute the capture action
+        if(isFeasible(gs)) {
+            targetCity.setActorID(this.unit.getActorID());
```

Comment:
the actor id of the city doesn't need to change (and setting it to the actor ID of the unit that captures it would be wrong anyway)